### PR TITLE
Refactors the API's demand editing functions

### PR DESCRIPTION
**Pull request by LRossman**

### CODE REVIEW

File: src/epanet.c [hunk 1]
```diff
@@ -2830,22 +2888,19 @@ int DLLEXPORT EN_getbasedemand(EN_Project p, int nodeIndex, int demandIndex,
...
-        list_node_t *lnode = get_nth_list(dlist, demandIndex);
-        if (!lnode)
-            return 253;
-        else
-            *baseDemand = get_base_demand(lnode) * p->Ucf[FLOW];
-    }
-    else *baseDemand = (double)(0.0);
+    // Locate target demand in node's demands list
+    d = finddemand(p->network.Node[nodeIndex].D, demandIndex);
+    if (d == NULL) return 253;
```
Comment (samhatchett) [inline src/epanet.c]: not sure if we ever settled on a code style guide... but these single-line conditional/returns are so frustrating when I want to place a breakpoint. I'm completely fine moving forward with this PR as-is, but let's make a determination about style soon 👍

**Comment by eladsal** [comment]
GH desktop won't allow me to use this PR, it reports :
image
I have no idea what is wrong, `dev` is up to date, any ideas?

**Comment by eladsal** [comment]
@LRossman I'm not sure how this is related to this PR but in order to have the old win build script run I had to add to the top of `rules.c` the following:
```
#ifdef _WIN32
#define snprintf _snprintf
#endif
```

**Comment by LRossman** [comment]
@eladsal the `CMakeLists `script includes the `-D_CRT_SECURE_NO_DEPRECATE` directive which allows its build to ignore any errors/warnings if `snprintf `is used instead of `_snprintf`. I assume that the old win build script didn't do that and  `snprintf` got flagged as an error by the VS 2010 compiler. The same `#define` for `snprintf` was already being used in the _epanet.c_ and _report.c_ modules.

**Pull request merged**